Materialize

eros2123 opened the pull request:
vxbn

petebacondarwin committed:
chore(docs): add `getComponentPath` helper service

petebacondarwin committed:
chore(docs): remove unused cookie service and module

petebacondarwin committed:
chore(docs): update deployments to use angular material and not bootstrap

gkalpak committed:
docs(app): minor refactoring of 'docs/app/src/*'

jesselpalmer committed:
chore(docs): replace "Edit in Plunker" button with Angular Material

Closes #10305

gkalpak committed:
docs(app): refactor the Docs App based on the MD prototype

I incorporated the prototype detailed in https://github.com/gkalpak/angular.js/issues/3.

Things that are different from the prototype (besides using actual data):

 * The left sidenav (toc) is not styled properly (or even readably).
 * The subheaders in the left sidenav are not "stickying" properly (only `ng` becomes sticky).
 * Icons are missing (since we are not using FontAwesome). E.g. search-fab, toc-toggle, results
 * In the right sidenav (search) there are some unwanted horizontal scrollbars.
   (This section probably needs rethinking anyway.)
 * I have implemented a `responsiveMenu` directive to show the options in the header-menus
   (Learn/Develop/Discuss). The purpose is to show the options in a "dropdown-ish" menu on larger
   screens and using a BottomSheet on smaller. At the moment a BottomSheet is used on every
   screen-size.

Notes:

 - Still no version-picker and breadcrumbs (I have left the original implementation of those two
   commented out).
 - I have replaced the previous docs.css with a new one containing the style of the prototype.
   I have renamed the old one to `docs_old.css` and left it there temporarily for quick reference.
   We need to clean up this and the rest of the old CSS files.
 - The main content area is totally unstyled and needs some love (a lot actually).

I have also "sprinkled" the code with a few more TODOs/FIXMEs.

(For more details on known issues and missing features take a look at
 https://github.com/gkalpak/angular.js/issues/3)

gkalpak committed:
docs(app): fix styling of left sidenav, add FontAwesome

Since the structure was a little different than the prototype,
classes/styles where not applied correctly to the links in the left
sidenav. This commit fixes the issue.

I also added FontAwesome, in order to be able to display icons until the
Material Design icon-font is released. Since this is a temporary meassure,
I didn't added as a bower dependency; just added a `font-awesome` folder
in assets and added `font-awesome/css/font-awesome.css` to the stylesheets
of each deployment (under `docs/config/services/deployments/`.

gkalpak committed:
docs(app): make tests pass

Due to the recent changes ragarding the Docs app "Materialization", some
tests where failing for the following reasons:

1. JQuery wasn't being copied to 'build/docs/components/'.
2. E2E examples 89 and 90 were relying on the Bootstrap's glyphicons.
   Temporarily replaced with FontAwesome icons; to be replaced with
   Material Design iconfont, when ready.
3. `ngMaterial` (and its various dependencies) weren't loaded by Karma
during unit testing (causing modules depending on it to fail). Added a
fake `ngMaterial` module (as a temporary solution).

jesselpalmer committed:
docs(app): matches anchor tags to bootstrap

jesselpalmer committed:
docs(app): fixes tabbing

joshkurz committed:
docs(app): adding a mobile first breadcrumb solution

Signed-off-by: Josh Kurz <jkurz25@gmail.com>

jesselpalmer committed:
docs(app): adds roboto font

jesselpalmer committed:
docs(app): updates typography

jesselpalmer committed:
docs(app): replaces font px w/ %

jesselpalmer committed:
docs(app): adds stylesheets to deployment files

jesselpalmer committed:
docs(app): deletes icon

We won't use the material icons manually. We will use font awesome until material icons are fully supported.

jesselpalmer committed:
docs(app): adds hr

jesselpalmer committed:
docs(app): removes material icon reference

Removes material icon reference until material icons are ready to be used.

jesselpalmer committed:
docs(app): makes minor updates

jesselpalmer committed:
docs(app): increments year

jesselpalmer committed:
docs(app): fixes material build

jkurz commented:
Is this something the team is still trying to achieve? Or is it not a priority anymore? If it is, then this PR should probably be closed and the branch should be given some more love.

matsko committed:
fix(ngAnimate): close follow-up class-based animations when the same class is added/removed when removed/added

This patch ensures that if the same CSS class is added/removed within a
follow-up digest then the previous class-based animation is cancelled
beforehand.

Closes #11717

maxthyen committed:
fix($http): do not modify the config object passed into $http short methods

Update $http's createShortMethods and createShortMethodsWithData
to extend an empty object instead of the passed-in config.
Previously, since $http was extending the passed-in config,
the changes to the config object persisted even after the call to $http's
get/post/etc. returned. This causes unexpected behavior if that
config object is reused in subsequent calls to $http.
The existing test in httpSpec was not properly testing this situation.

Closes: #11764

Commit:
feat(filterFilter): allow array like objects to be filtered

Throw error if filter is not used with an array like object.

Closes #11782
Closes #11787

matsko committed:
fix(ngAnimate): class-based animations must not set addClass/removeClass CSS classes on the element

With the abstraction system that ngAnimate uses, $animateCss internally
sets the provided `event` as a CSS class on the element. In this
situation the `addClass` and `removeClass` events on the element as a
CSS class. This should not happen with class-based animations and this
feature is unnecessary and has now been removed.

Closes #11810

jesselpalmer committed:
adds angulario resources

Narretz committed:
docs(migration): general style improvements

Closes #11849

matsko committed:
fix(ngAnimate): ensure anchored animations remove the leave element at correct time

Due to a mismatch of where the `options.domOperation` value was stored,
the element involved in the `leave` animation for an anchored animation
session was not removed as soon as the leave animation ended. This
resulted in a pending element persisting within the DOM until all
animations involved in the associated anchor animation were complete.
This patch fixes this issue.

Closes #11850

matsko committed:
fix($animateCss): ensure that custom durations do not confuse the gcs cache

Closes #11723
Closes #11852

petebacondarwin committed:
fix(ngAnimate): throw an error if a callback is passed to animate methods

As of bf0f5502b1bbfddc5cdd2f138efd9188b8c652a9 (released in 1.3.0) it is no longer
valid to pass a callback to the following functions: `enter`, `move`, `leave`, `addClass`,
`removeClass`, `setClass` and `animate`.

To prevent confusing error messages, this change asserts that this parameter is
not a function.

Closes #11826
Closes #11713

petebacondarwin committed:
docs(error//nocb): add error doc for invalid parameter

petebacondarwin committed:
docs(CHANGELOG): update to 1.4.0-rc.2

divercraig committed:
fix(select): prevent unknown option being added to select when bound to null property

If a select directive was bound, using ng-model, to a property with a value of null this would
result in an unknown option being added to the select element with the value "? object:null ?".
This change prevents a null value from adding an unknown option meaning that the extra option is
not added as a child of the select element.

Since select (without ngOptions) can only have string value options then `null` was never a
viable option value, so this is not a breaking change.

Closes #11872
Closes #11875

matsko committed:
fix($animate): accept unwrapped DOM elements as inputs for enter + move

Prior to this fix the $animate.enter() and $animate.move() events caused
an error when a parent or after element was provided that was not
already wrapped as a jqLite element. This patch ensures that both
wrapped and unwrapped DOM nodes are allowed.

Closes #11848

matsko committed:
fix(ngAnimate): ensure that repeated structural calls during pre-digest function

Prior to this fix if `$animate.enter()` or `$animate.leave()` was called
before a digest was issued then the element may not be cancelled early
enough. This fix ensures that the previous structural animation is
cancelled immediately when a follow-up animation is kicked off.

Closes #11867

petebacondarwin committed:
docs(select): provide more info and example about non-string options

See #11890

matsko committed:
fix(ngAnimate): ensure that cancelled class-based animations are properly cleaned up

Prior to this fix if the same CSS class was added and removed quickly
then the element being animated would be left with a stale cache of the
cancelled out animation. This would then result in follow-up animations
being added to the previous animation which would then never run. A
stale cache was to blame for that. This patch takes care of this issue.

Closes #11652

matsko committed:
fix(ngAnimate): ensure nested class-based animations are spaced out with a RAF

Prior to this fix any nested class-based animations (animations that are
triggered with addClass/removeClass or ngClass) would cancel each other
out when nested in DOM structure. This fix ensures that the nested
animations are spaced out with sequenced RAFs so that parent CSS classes
are applied prior to any ancestor animations that are scheduled to run.

Closes #11812

juliemr committed:
chore(test): update protractor to 2.1

gkalpak committed:
docs(ngPattern): add note about using the `g` flag

Add a note to point out that using the `g` flag on the validation RegExp,
will cause each search to start at the index of the last search's match,
thus not taking the whole input value into account.

Closes #11917
Closes #11928

matsko committed:
revert: fix(ngAnimate): throw an error if a callback is passed to animate methods

This reverts commit 9bb4d6ccbe80b7704c6b7f53317ca8146bc103ca.

petebacondarwin committed:
test(matchers): fix "not" string for `toHaveClass` matcher

matsko committed:
fix($animate): ignore invalid option parameter values

Prior to this fix there was another patch that threw an exception if the
provided options value was not of an object type. While this is correct
in terms of logic, it caused issues with plugins and tools that are
designed to work with multiple version of Angular. This fix ensures that
these plugins work since an invalid options value is ignored by
`$animate`.

Closes #11826

matsko committed:
docs(CHANGELOG): add changes for 1.4.0

jesselpalmer committed:
angular.io styles added

jesselpalmer committed:
removes unneeded angular.io files

jesselpalmer committed:
add css styles to deployment

jesselpalmer committed:
docs(app): merges master into materialize branch

jesselpalmer committed:
Merge pull request #11955 from jesselpalmer/updates-branch-with-master

Updates branch with master

lgalfaso commented:
In this state, it looks impossible to understand what the fix is. Please rebase this or just open a new PR with the fix
